### DISCUSSION TITLE
Headless multiple custom storefronts

### DISCUSSION THREAD

**Discussion by miksony** [Shopify/storefront-api-feedback#210] [Help] [answered]
Hi everyone!

We have been using Storefront API for a while and we are very satisfied with the stability and speed it provides.

Our use case is that we have one Shopify store with a Headless sales channel and multiple custom storefronts.

**_Is there a way to limit products and collections results for a custom storefront?_** 

In the products and collections admin section, there is a way to limit by a sales channel (which can be Headless) by not by a custom storefront individually.
Is there maybe another Shopify feature that would provide this to us and fit our use case (we are also using Plus)? 

Currently workaround for this would be that we assign some metafield which we provide in a query.

**Comment by johncraigcole** [comment, marked as answer]
Key @miksony 👋 . Today, this would only be possible via query as you noted. Just be cognizant that something like[ products query](https://shopify.dev/docs/api/storefront/2023-07/queries/products#argument-products-query) on it's own does not support filtering by metafield. Metafield filtering is possible in the [context of collection](https://shopify.dev/docs/api/storefront/2023-07/input-objects/ProductFilter), and with [search](https://shopify.dev/docs/api/storefront/2023-07/queries/search). Using a tag might be a better option depending on the type of queries being performed.

I'll be sure to pass this along to the product teams so they're aware of  the use case.
